src/CambrianRouter.sol

```diff
@@ -1,14 +1,28 @@
 pragma solidity ^0.8.20;
 
-import {Request} from "./Cambrian.sol";
+import {Request, Register} from "./Cambrian.sol";
+
+struct wrapper {
+    string val;
+    bool isValue;
+}
 
 contract CambrianRouter {
+    mapping(address => wrapper) public queries;
```

Comment:
I think wrapper is not needed, we should be able to check for queries[addr] != ""

```suggestion
contract CambrianRouter {
    mapping(address => string) public queries;
```

Reply:
Check this question for how to check if mapping value is set for a string (cast to bytes and check length)

https://ethereum.stackexchange.com/questions/5683/what-is-the-zero-value-for-a-string